src/interface.c

```diff
@@ -297,6 +304,11 @@ static int interface_read (void)
 		if (kstat_read (kc, ksp[i], NULL) == -1)
 			continue;
 
+		if (unique_name)
+			snprintf(iname, sizeof(iname), "%s_%d_%s", ksp[i]->ks_module, ksp[i]->ks_instance, ksp[i]->ks_name);
```

Comment:
Please use `ssnprintf()` (note the extra "s"): that version will ensure that `iname` is null-terminated.